simulator/infra/config/t/unit_test.cpp

```diff
@@ -0,0 +1,252 @@
+/*
+* This is an open source non-commercial project. Dear PVS-Studio, please check it.
+* PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+*/
+/**
+ * Unit tests for configuration parser
+ * @author Denis Los
+ */
+
+
+// generic C
+#include <cstdlib>
+#include <cassert>
+
+
+// Google test library
+#include <gtest/gtest.h>
+
+
+#include "../config.h"
+
+
+namespace config {
+static RequiredValue<std::string> binary_filename = { "binary,b", "input binary file"};
+static RequiredValue<uint64> num_steps = { "numsteps,n", "number of instructions to run"};
+
+static Value<bool> disassembly_on = { "disassembly,d", false, "print disassembly"};
+static Value<bool> functional_only = { "functional-only,f", false, "run functional simulation only"};
+
+namespace utils {    
+template <typename T, std :: size_t size>
+constexpr std :: size_t countof( T (&)[ size]) { return size; } 
```

Comment:
What's the point of copying and not just including "macro.h" (+I'm sure it is already included to the config.h)?